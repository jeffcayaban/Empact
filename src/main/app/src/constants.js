export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;