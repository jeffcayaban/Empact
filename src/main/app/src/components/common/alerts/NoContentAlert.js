import React from 'react';
import { Alert } from "react-bootstrap";
import { PETITION } from "../../../constants";

/**
 * Is the alert for when a user is attempting to create an argument of their own petition.
 */

class NoContentAlert extends React.PureComponent {

    render() {
        const displayedContentType = this.props.contentType === PETITION ? "petitions" : "arguments";

        return (
            <div style={{ margin: '15px' }}>
                <Alert bsStyle="warning">{`There are no ${displayedContentType} here!`}</Alert>
            </div>
        );
    }
}

export default NoContentAlert;