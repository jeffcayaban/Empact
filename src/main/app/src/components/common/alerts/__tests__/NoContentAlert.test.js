import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import NoContentAlert from "../NoContentAlert";
import {ARGUMENT, PETITION} from "../../../../constants";

describe('NoContentAlert', () => {

    it('should render successfully for petition', () => {
        const component = shallow(<NoContentAlert contentType={PETITION} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render successfully for argument', () => {
        const component = shallow(<NoContentAlert contentType={ARGUMENT} />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
