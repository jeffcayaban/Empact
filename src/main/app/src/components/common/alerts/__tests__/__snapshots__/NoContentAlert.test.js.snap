// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoContentAlert should render successfully for argument 1`] = `
<div
  style={
    Object {
      "margin": "15px",
    }
  }
>
  <Alert
    bsClass="alert"
    bsStyle="warning"
    closeLabel="Close alert"
  >
    There are no arguments here!
  </Alert>
</div>
`;

exports[`NoContentAlert should render successfully for petition 1`] = `
<div
  style={
    Object {
      "margin": "15px",
    }
  }
>
  <Alert
    bsClass="alert"
    bsStyle="warning"
    closeLabel="Close alert"
  >
    There are no petitions here!
  </Alert>
</div>
`;
